grails-app/controllers/au/org/ala/ecodata/PermissionsController.groovy

```diff
@@ -1048,6 +1048,25 @@ class PermissionsController {
         }
     }
 
+    def isUserMemberOfProject() {
+        String userId = params.userId
+        String projectId = params.projectId
+        String roles = params.roles ?: "admin,caseManager,moderator,editor,projectParticipant"
```

Comment:
Rather than hard-coding the strings here, it would be better to use the AccessLevel enum to derive them.  It would also be more http idomatic to use params.getList() and send it from biocollect in the form &roles=admin&roles=moderator etc.

Reply:
WIthout understanding how it's used in biocollect, an alternative would be to use "findUserPermission(userId, projectId)" and do the role processing on the client side.

Reply:
Changed.
Doing on client side was causing BioCollect to be really slow on projects with large number of members. Moved BioCollect functionality to  ecodata to address this issue.